lib/sift/value_parser.rb

```diff
@@ -57,5 +60,20 @@ def boolean_value
         ActiveRecord::Type::Boolean.new.type_cast_from_user(value)
       end
     end
+
+    def normalize_date_range
+      from_date_string, end_date_string = value.split("...")
+      return unless end_date_string
+
+      parsed_dates = [from_date_string, end_date_string].map do |date_string|
+        begin
+          DateTime.parse(date_string.to_s)
+        rescue StandardError
+          date_string
+        end
+      end
+
+      @value = parsed_dates.join("...")
```

Comment:
Do we need to modify @value?

Maybe better to just return and array of DateTimes or return a range directly
`Range.new(parsed_dates)`

Reply:
refactored a little bit to make this method return a value

regarding
> Maybe better to just return and array of DateTimes or return a range directly
> Range.new(parsed_dates)

I am not sure it worth doing at the moment
1. Since it's a change of value type we have to very thoroughly check that all "consumers" of this value will handle it appropriately. Not sure if tests show if something went wrong. Potentially bugs may appear
2. Don't see what particular benefit we will get at the moment.